python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -327,30 +359,41 @@ def _forward_mouse_event(self, mouse_event, index):
         QtGui.QApplication.sendEvent(widget, forward_event)
 
 
-def _set_widget_value(widget, model_index):
-    """
-    Updates the supplied widget with data from the supplied model index.
+    def _set_widget_value(self, widget, model_index):
+        """
+        Updates the supplied widget with data from the supplied model index.
 
-    :param widget: The widget to set the value for
-    :type parent: :class:`~PySide.QtGui.QWidget`
-    :param model_index: The index of the model where the data comes from
-    :type model_index: :class:`~PySide.QtCore.QModelIndex`
-    """
+        :param widget: The widget to set the value for
+        :type parent: :class:`~PySide.QtGui.QWidget`
+        :param model_index: The index of the model where the data comes from
+        :type model_index: :class:`~PySide.QtCore.QModelIndex`
+        """
+
+        src_index = _map_to_source(model_index)
+        if not src_index or not src_index.isValid():
+            # invalid index, do nothing
+            return
 
-    src_index = _map_to_source(model_index)
-    if not src_index or not src_index.isValid():
-        # invalid index, do nothing
-        return
+        # special case for image fields in the ShotgunModel. The SG model has
+        # the ability to pre-query thumbnails for entities for efficiency. If
+        # this is the image field for an entity in the SG model, we can make use
+        # of the potentially pre-queried image available in the first column.
+        if self._source_is_sg_model(model_index) and widget._field_name == "image":
```

Comment:
Same comment for using it with something else than a ShotgunModel, and having a `is_image` property or method.